src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -217,6 +227,40 @@ public void start(final Map<String, String> props) {
     return null;
   }
 
+  private Optional<String> convertToJson(Optional<BsonDocument> jsonDocumentraw, JsonWriterSettings jsonOptions) {
+        if(jsonOptions!=null && jsonDocumentraw.isPresent()){
+          return Optional.of(jsonDocumentraw.get().toJson(jsonOptions));
+        }
+        else if(jsonDocumentraw.isPresent()){
+          return Optional.of(jsonDocumentraw.get().toJson());
+        }
+        else{
+          return Optional.empty();
+        }
+  }
+
+  private JsonWriterSettings handleJsonOptions() throws CloneNotSupportedException {
+    switch (sourceConfig.getJsonType()) {
+      case EXTENDED:
+        return (JsonWriterSettings.builder()
+                .outputMode(JsonMode.EXTENDED)
+                .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                .build());
+      case  SHELL:
+        return (JsonWriterSettings.builder()
+                .outputMode(JsonMode.SHELL)
+                .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                .build());
+      case RELAXED:
+        return (JsonWriterSettings.builder()
+                .outputMode(JsonMode.RELAXED)
+                .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+                .build());
```

Comment:
There's quite a lot of duplication in this method. Can we improve that?

Reply:
i look a bit around but didn't find something easier. We need to do it three times do you think it's to much ?

Reply:
Hint: [Java supports fall-through when you have no `break`](https://stackoverflow.com/a/48233495/65060).

Reply:
thanks i have changed the switch !